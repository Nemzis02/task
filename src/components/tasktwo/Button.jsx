import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const StyledButton = styled.button`
  width: 105px;
  height: 40px;
  font-size: 16px;
  font-weight: bold;
  color: #4b79b4;
  background-color: #cae5f5;
  border: none;
  outline: none;
  border-radius: 6px;
  cursor: pointer;

  &:disabled {
    color: #97a0ac;
    cursor: default;
  }

  ${({ $variant }) => $variant === 'secondary' && css`
      border: 1px solid #4b79b4;
      background-color: #ffffff;
    `};
`;

const Button = (props) => {
  const {
    variant,
    type,
    text,
    className,
    onClick,
    disabled
  } = props;

  return (
    <StyledButton
      $variant={variant}
      type={type}
      className={className}
      onClick={onClick}
      disabled={disabled}
    >
      {text}
    </StyledButton>
  );
};

Button.defaultProps = {
  variant: 'main',
  type: 'button',
  text: 'Yes',
  className: '',
  disabled: false
};

Button.propTypes = {
  variant: PropTypes.string,
  type: PropTypes.string,
  text: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool
};

export default Button;
