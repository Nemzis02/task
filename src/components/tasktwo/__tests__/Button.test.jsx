import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from '../Button';

describe('Button Component', () => {
  test('Should be rendered correctly', () => {
    const { container } = render(<Button text="Hello" onClick={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  test('Should render text passed via props', () => {
    const text = 'Click';

    render(<Button text={text} onClick={() => {}} />);

    const result = screen.getByText(text);

    expect(result).toBeInTheDocument();
  });

  test('Should call onClick callback when it is clicked', () => {
    const buttonText = 'Do';
    const onClick = jest.fn();

    render(<Button text={buttonText} onClick={onClick} />);

    fireEvent.click(screen.getByText(buttonText));

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
