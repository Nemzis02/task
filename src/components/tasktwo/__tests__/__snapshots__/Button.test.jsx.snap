// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should be rendered correctly 1`] = `
<div>
  <button
    class="Button__StyledButton-ehu7d2-0 knoOUb"
    type="button"
  >
    Hello
  </button>
</div>
`;
